source/deployment/deployment.md

```diff
@@ -48,14 +47,37 @@ Typically a "Mattermost unreachable" error message will be displayed warning use
 
-To deploy a system outside your firewall, by opening standard ports such as 80 and 443, we recommend: 
+- In general, Mattermost recommends access via a Virtual Private Network client using at least a secondary authentication option, such as [OpenVPN](https://openvpn.net/), in addition to Mattermost authentication. 
+- For deployments accessible from the open internet, it is highly recommended to have dedicated staff managing system security and maintaining security updates in addition to enabling 2FA features in Mattermost Enterprise Edition. For non-enterprise deployments VPN is recommended. 
 
-1. Ensuring that an IT admin is subscribed to the [Mattermost security bulletin](https://about.mattermost.com/security-bulletin/) and able to apply any new security updates. 
-2. Upgrading to Mattermost Enterprise Edition for the ability to use multi-factor authentication without the MFA-enabled VPN client required for Team Edition deployments.
+While there are a multitude of different deployment configurations, the following outline popular configurations: 
+
+- **1. Private Network only with MFA (TE & EE)** - Access to Mattermost only available on private network. Multi-factor authentication includes securing access to computing devices along with physical security within buildings. Mobile web view used in place of Native Mobile Applications. Used in very high security, on-premises deployments. 
+- **2. Private Network or VPN with MFA (TE & EE)** - Private network deployment in 1) plus the option of accessing Mattermost using Virtual Private Network ("VPN") client supporting a secondary authentication method plus Mattermost's own authentication.  
+- **3. Private Network or VPN, including private Mobile Apps, with MFA (TE & EE)** - Deployment in 2) with privately-compiled mobile solution including both mobile applications and mobile push notification services compiled by the organization with organization-controlled keys or certificates (versus using public Mobile Apps, which use keys and certificates controlled by Mattermost, Inc.). 
+- **4. Private Network or VPN with MFA plus private Mobile Apps with EMM (TE & EE)** - Deployment in 3) with the addition of an enterprise mobility management solution such as Mobile Iron, Blackberry Dynamics or AirWatch for access to mobile apps in place of VPN clients. 
+- **5. Private Network or VPN with MFA, plus public Mobile Apps with 2FA (EE-only)** - Deployment in 2) plus use of publicly available mobile apps in iTunes or Google Play with the additional of MFA available in Enterprise Edition via single-sign-on providers like Okta, OneLogin, and SAML-based options, or user credentials with the addition of Google Authenticator. Recommended only for enterprises resourced to provide appropriate network security and monitoring for systems exposed to open internet. For non-enterprise deployments, VPN with MFA is recommended (see above).
```

Comment:
Also on second thought, we shouldn't need to list SSO providers here. If that list ever changes, we'll need to remember to update it so it may be enough to say "via SAML single sign-on providers or..."